### PR TITLE
[Ready to Merge] fldeff_strength.s, fldeff_sweetscent.s

### PR DESCRIPTION
**Pull request by KDSKardabox**

### CODE REVIEW

File: ld_script.txt [hunk 1]
```diff
@@ -193,6 +193,7 @@ SECTIONS {
         asm/pokedex_cry_screen.o(.text);
         src/coins.o(.text);
         src/landmark.o(.text);
+		src/fldeff_strength.o(.text);
```
Comment (ProjectRevoTPP) [inline ld_script.txt]: tabs vs spaces

File: asm/fldeff_strength.s [hunk 2]
```diff
@@ -5,51 +5,51 @@
...
+//	ldr r1, =gFieldEffectArguments
+//	lsls r0, 24
+//	lsrs r0, 24
+//	str r0, [r1]
+//	ldr r0, =FieryPath_EventScript_2908FD
+//	bl ScriptContext1_SetupScript
+//	pop {r0}
+//	bx r0
+//	.pool
+//	thumb_func_end sub_8145E0C
```
Comment (ProjectRevoTPP) [inline asm/fldeff_strength.s]: decompiled assembly should just be deleted

File: src/fldeff_strength.c [hunk 3]
```diff
@@ -0,0 +1,109 @@
+#include "global.h"
+#include "field_effect.h"
+#include "rom6.h"
+#include "party_menu.h"
+
```
Comment (ProjectRevoTPP) [inline src/fldeff_strength.c]: space?

File: ld_script.txt (hunk 1 above)
Comment (KDSKardabox) [reply ld_script.txt]: It was a dirty commit to just claim the file. But now the file has been fully decomped with all of these points already kept in mind.

File: src/fldeff_sweetscent.c [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+#include "script.h"
+#include "sound.h"
+#include "sprite.h"
+#include "task.h"
+#include "wild_encounter.h"
+
+extern void sub_81BE72C(void);
+extern void sub_81BE6B8(void);
+extern void sub_80AC3D0(void);
+extern void sub_80AC3E4(void);
```
Comment (Diegoisawesome) [inline src/fldeff_sweetscent.c]: Put these in the appropriate headers and include them

File: src/fldeff_sweetscent.c [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    gTasks[taskId].data[8] = (u32)sub_8159F5C >> 16;
+    gTasks[taskId].data[9] = (u32)sub_8159F5C;
+    return FALSE;
+}
+
+void sub_8159F5C(void)
+{
+    u8 taskId;
+
+    PlaySE(0xEC);
```
Comment (Diegoisawesome) [inline src/fldeff_sweetscent.c]: Use constant

File: src/fldeff_sweetscent.c [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    CpuFastSet(gPlttBufferUnfaded, gPaletteDecompressionBuffer, 0x100);
+    CpuFastSet(gPlttBufferFaded, gPlttBufferUnfaded, 0x100);
+    BeginNormalPaletteFade(~(1 << (gSprites[GetPlayerAvatarObjectId()].oam.paletteNum + 16)), 4, 0, 8, 0x1F);
```
Comment (Diegoisawesome) [inline src/fldeff_sweetscent.c]: Use RGB_RED constant

File: src/fldeff_sweetscent.c [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    gTasks[taskId].data[0] = 0;
+    FieldEffectActiveListRemove(FLDEFF_SWEET_SCENT);
+}
+
+void sub_8159FEC(u8 taskId)
+{
+    if (!gPaletteFade.active)
+    {
+        sub_81BE72C();
+        BlendPalettes(64, 8, 0x1F);
```
Comment (Diegoisawesome) [inline src/fldeff_sweetscent.c]: Change 64 to 0x00000040 and use RGB_RED constant

File: src/fldeff_sweetscent.c [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            gTasks[taskId].data[0] = 0;
+            if (SweetScentWildEncounter() == TRUE)
+            {
+                DestroyTask(taskId);
+            }
+            else
+            {
+                gTasks[taskId].func = sub_815A090;
+                BeginNormalPaletteFade(~(1 << (gSprites[GetPlayerAvatarObjectId()].oam.paletteNum + 16)), 4, 8, 0, 0x1F);
```
Comment (Diegoisawesome) [inline src/fldeff_sweetscent.c]: Use RGB_RED constant

File: src/fldeff_strength.c [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+#include "field_effect.h"
+#include "party_menu.h"
+#include "rom6.h"
+#include "script.h"
+#include "task.h"
+
+void sub_8145E0C(void);
+void sub_8145E74(void);
+
+bool8 hm_prepare_rocksmash(void)
```
Comment (Diegoisawesome) [inline src/fldeff_strength.c]: Rename this function (globally) to SetUpFieldMove_Strength

File: src/fldeff_strength.c [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    return FALSE;
+}
+
+void sub_8145E0C(void)
+{
+    gFieldEffectArguments[0] = GetCursorSelectionMonId();
+    ScriptContext1_SetupScript(FieryPath_EventScript_2908FD);
+}
+
+bool8 sub_8145E2C(void)
```
Comment (Diegoisawesome) [inline src/fldeff_strength.c]: Rename this function (globally) to FldEff_UseStrength

Comment (KDSKardabox) [reply src/fldeff_strength.c]: all done

**Pull request merged**